src/httpmessage.rs

```diff
@@ -200,7 +200,7 @@ pub trait HttpMessage: Sized {
     /// # fn main() {}
     /// ```
     fn json<T: DeserializeOwned>(&self) -> JsonBody<Self, T> {
-        JsonBody::new(self)
+        JsonBody::new::<()>(self, None)
```

Comment:
is adding `::<()>` necessary?

Reply:
Without it I get:

```
error[E0282]: type annotations needed
   --> src/httpmessage.rs:203:9
    |
203 |         JsonBody::new(self, None)
    |         ^^^^^^^^^^^^^ cannot infer type for `S`
```

Is there something else I can do to help the compiler out there?